pkg/contextstore/metadata_test.go

```diff
@@ -0,0 +1,110 @@
+package contextstore
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+var testMetadata = ContextMetadata{
+	Metadata: map[string]interface{}{
+		"baz": "foo",
+	},
+	Endpoints: map[string]EndpointMetadata{
+		"test-ep": {
+			"foo": "bar",
+		},
+	},
+}
+
+func TestMetadataMarshalling(t *testing.T) {
+	var ctxMeta ContextMetadata
+	bytes, err := json.Marshal(&testMetadata)
+	assert.NilError(t, err)
+	err = json.Unmarshal(bytes, &ctxMeta)
+	assert.NilError(t, err)
+	assert.DeepEqual(t, ctxMeta, testMetadata)
+}
+
+func TestMetadataGetNotExisting(t *testing.T) {
+	testDir, err := ioutil.TempDir("", t.Name())
+	assert.NilError(t, err)
+	defer os.RemoveAll(testDir)
+	testee := metadataStore{root: testDir}
+	_, err = testee.get("noexist")
+	assert.Equal(t, true, os.IsNotExist(err))
+}
+
+func TestMetadataCreateGetRemove(t *testing.T) {
+	testDir, err := ioutil.TempDir("", t.Name())
+	assert.NilError(t, err)
+	defer os.RemoveAll(testDir)
+	testee := metadataStore{root: testDir}
+	expected2 := ContextMetadata{
+		Metadata: map[string]interface{}{
+			"baz": "foo",
+		},
+		Endpoints: map[string]EndpointMetadata{
+			"test-ep": {
+				"foo": "bar",
+			},
+			"test-ep2": {
+				"foo": "bar",
+			},
+		},
+	}
+	err = testee.createOrUpdate("test-context", testMetadata)
```

Comment:
AIUI this is supposed to persist into `testDir`. Is it worth checking that has happened by defining a new store on the same root and checking this stuff is still there?

Perhaps the implementation means that isn't needed.

Reply:
the metadataStore implementation is stateless. I added your suggestion to make sure it stays that way